### PR TITLE
[SYCL] Add local_accessor iterator tests

### PR DESCRIPTION
**Pull request by KornevNikita**
sycl patch: https://github.com/intel/llvm/pull/6692

### CODE REVIEW

File: SYCL/Basic/accessor/accessor.cpp [hunk 1]
```diff
@@ -787,5 +787,292 @@ int main() {
     }
   }
 
+  // Accessor begin/end member functions
```
Comment (AlexeySachkov) [inline SYCL/Basic/accessor/accessor.cpp]: I assume that those methods should also work on host. Therefore, I suggest to refactor the test a bit: let's introduce a few function templates, which would accept accessor iterators and do some operations with them. For example:
- `modify`, which accepts non-constant pair of iterators and modifies each element (multiple by 2 or add 1)
- `copy`, which accepts a pair of constant iterators and a non-const iterator: it copies from `const` to non-`const`

For those two functions you can easily write verification functions and prepare a bunch of test cases for both host and device.

Comment (KornevNikita) [reply SYCL/Basic/accessor/accessor.cpp]: Does it look better now? 328e794ae00ffe9b94e197c691e9eef1a1f03699

Comment (AlexeySachkov) [reply SYCL/Basic/accessor/accessor.cpp]: I think that we can go further. For example, "0-dim accessor iteration", "Simple iteration through the accessor", "Reverse iterator" and even "3-dim accessor simple iteration" look almost the same: we could create another function, which would accept a `buffer` with input data, then create queue, submit a kernel, test new methods at host and then perform a validation. The function should probably also accept `useReverse` bool to pass it further.

This way, you should be able to test several use-cases with just 4 calls, significantly reducing the amount of duplicated code.

The same can be applied for copy use-cases.

Comment (KornevNikita) [reply SYCL/Basic/accessor/accessor.cpp]: Okay, let's go further

Comment (KornevNikita) [reply SYCL/Basic/accessor/accessor.cpp]: @AlexeySachkov round 2 :) Does it look better now?

**Review by AlexeySachkov — DISMISSED** [review]
That looks much better to me, thanks!

**Comment by KornevNikita** [comment]
Re-run tests

**Review by AlexeySachkov — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**